api/src/services/fields.ts

```diff
@@ -412,15 +412,21 @@ export class FieldsService {
 			if (hookAdjustedField.schema) {
 				const existingColumn = await this.schemaInspector.columnInfo(collection, hookAdjustedField.field);
 
+				if (hookAdjustedField.schema?.is_nullable === true && existingColumn.is_primary_key) {
+					throw new InvalidPayloadError({ reason: 'Primary key cannot be null' });
+				}
+
 				// Sanitize column only when applying snapshot diff as opts is only passed from /utils/apply-diff.ts
 				const columnToCompare =
 					opts?.bypassLimits && opts.autoPurgeSystemCache === false ? sanitizeColumn(existingColumn) : existingColumn;
 
 				if (!isEqual(columnToCompare, hookAdjustedField.schema)) {
 					try {
-						await this.knex.schema.alterTable(collection, (table) => {
-							if (!hookAdjustedField.schema) return;
-							this.addColumnToTable(table, field, existingColumn);
+						await this.knex.transaction(async (trx) => {
+							await trx.schema.alterTable(collection, async (table) => {
+								if (!hookAdjustedField.schema) return;
+								this.addColumnToTable(table, field, existingColumn);
+							});
```

Comment:
Using transaction for item update is optional, but i feel we should to avoid these kind of inconsistencies, in this specific case, knex was generating a series of DDL queries and it was failing while setting nullable to PK, but before that default(auto increment) was removed, since it was not a transaction, statements(remove default in this case) was not rolled back.